src/app/views/icons/service/icon.service.ts

```diff
@@ -1,45 +1,28 @@
-import {
-    Injectable
-} from '@angular/core';
-import { Http } from '@angular/http';
-import { RouteResolver } from '../../../resolve/route.resolver';
+import { Injectable } from '@angular/core';
 import { IconInterface } from '../icon-item-component/icon-interface';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Injectable()
-export class iconService
+export class IconService
```

Comment:
```suggestion
export class IconApiService
```

Entsprechend auch die Datei umbenennen